pkg/webui/console/views/application-general-settings/index.js

```diff
@@ -77,14 +76,15 @@ export default class ApplicationGeneralSettings extends React.Component {
   }
 
   async handleSubmit (values, { resetForm }) {
-    const { application } = this.props
+    const { application, updateApplication } = this.props
 
     await this.setState({ error: '' })
 
     const changed = diff(application, values)
+    const { ids: { application_id }} = application
+
     try {
-      const { ids: { application_id }} = application
-      await api.application.update(application_id, changed)
+      await updateApplication(application_id, changed)
```

Comment:
Using the update action here [will cause issues when you remove a value](https://github.com/TheThingsNetwork/lorawan-stack/issues/723#issuecomment-505826575) that was previously filled (e.g. application name). The result of the API request is not able to act as a proper patch for the reducer, because the removed field is not present.
Meaning: the field will be properly updated, but the store value remains the same. 

I solved this [in my PR](https://github.com/TheThingsNetwork/lorawan-stack/pull/855) by using the `updateSuccess` action instead, after issuing the API request myself.

Reply:
this doesnt seem right. please see my implementation